ldms/src/sampler/lustre_mdt/lustre_mdt_general.c

```diff
@@ -227,77 +246,67 @@ ldms_set_t mdt_general_create(const char *producer_name,
         return set;
 }
 
-static void mdt_md_stats_sample(const char *stats_path,
-                                   ldms_set_t general_metric_set)
+static int mdt_md_stats_sample(const char *stats_path,
+			       ldms_set_t metric_set)
 {
-        FILE *sf;
-        char buf[512];
-        char str1[64+1];
-
-        sf = fopen(stats_path, "r");
-        if (sf == NULL) {
-                log_fn(LDMSD_LWARNING, SAMP" file %s not found\n",
-                       stats_path);
-                return;
-        }
+	FILE *sf;
+	char buf[512];
+	int rc = 0;
+
+	sf = fopen(stats_path, "r");
+	if (sf == NULL) {
+		log_fn(LDMSD_LWARNING, ": file %s not found\n",
+			 stats_path);
+		return ENOENT;
+	}
 
-        /* The first line should always be "snapshot_time"
+	/* The first line should always be "snapshot_time"
            we will ignore it because it always contains the time that we read
            from the file, not any information about when the stats last
            changed */
-        if (fgets(buf, sizeof(buf), sf) == NULL) {
-                log_fn(LDMSD_LWARNING, SAMP" failed on read from %s\n",
-                       stats_path);
-                goto out1;
-        }
-        if (strncmp("snapshot_time", buf, sizeof("snapshot_time")-1) != 0) {
-                log_fn(LDMSD_LWARNING, SAMP" first line in %s is not \"snapshot_time\": %s\n",
-                       stats_path, buf);
-                goto out1;
-        }
+	if (fgets(buf, sizeof(buf), sf) == NULL) {
+		log_fn(LDMSD_LWARNING, ": failed on read from %s\n",
+			 stats_path);
+		rc = ENOMSG;
+		goto out1;
+	}
+	if (strncmp("snapshot_time", buf, sizeof("snapshot_time")-1) != 0) {
+		log_fn(LDMSD_LWARNING, ": first line in %s is not \"snapshot_time\": %s\n",
+			 stats_path, buf);
+		rc = ENOMSG;
+		goto out1;
+	}
 
-        ldms_transaction_begin(general_metric_set);
```

Comment:
lutre_mdt backport needs to retain the ldms_transaction_begin/end

Reply:
Fixed.